app/src/test/java/com/android/sample/model/map/LocationViewModelTest.kt

```diff
@@ -120,4 +121,34 @@ class LocationViewModelTest {
     // Then
     assertEquals(null, viewModel.currentLocation.first())
   }
+
+  @Test
+  fun `getDistanceFromCurrentLocation should return null when current location is null`() =
+      runTest {
+        // Given
+        viewModel.setCurrentLocation(null)
+        val activityLocation = Location(1.0, 2.0, "Test Location")
+
+        // When
+        val distance = viewModel.getDistanceFromCurrentLocation(activityLocation)
+
+        // Then
+        assertEquals(null, distance)
+      }
+
+  @Test
+  fun `getDistanceFromCurrentLocation should return distance between current location and activity location`() =
+      runTest {
+        // Given
+        val currentLocation = Location(46.518831258, 6.559331096, "EPFL")
+        viewModel.setCurrentLocation(currentLocation)
+        val activityLocation = Location(46.5375, 6.573611, "Epenex")
+        // When
+        val distance = viewModel.getDistanceFromCurrentLocation(activityLocation)
+        // Then
+        assertNotNull(distance)
+        if (distance != null) {
+          assertEquals(2.34f, distance, 0.05f)
+        }
+      }
```

Comment:
The delta is really important here ! Calculations may differ depending on formulas. Here you gave a 50m precision, which is adapted to our use case. Perfect!